.github/workflows/test-integration-postgres-redis-rabbit-matrix.yml

```diff
@@ -0,0 +1,171 @@
+    outputs:
+      matrix-json: ${{ steps.setup-matrix.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup matrix
+        id: setup-matrix
+        uses: OpenPhone/gha/.github/actions/setup-matrix@zach-voice-8352-leverage-github-action-matrix-for-jest-tests
+        with:
+          shards: ${{ inputs.shards }}
+
+  test-integration-postgres-rabbit-redis-matrix:
+    needs: [ setup-matrix ]
+    runs-on: ubuntu-latest
+    timeout-minutes: ${{ inputs.timeout-minutes }}
+
+    services:
+      postgres-svc:
+        image: postgres:13
+        env:
+          POSTGRES_USER: ${{ inputs.postgres_user }}
+          POSTGRES_PASSWORD: ${{ inputs.postgres_password }}
+          POSTGRES_DB: ${{ inputs.postgres_database }}
+        ports:
+          - 5432:5432
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+      redis-svc:
+        image: redislabs/rejson:latest
+        ports:
+          - 6379:6379
+        options: >-
+          --health-cmd "redis-cli ping"
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+      rabbitmq:
+        image: rabbitmq:3.9.7-management
+        env:
+          RABBITMQ_DEFAULT_USER: openphone
+          RABBITMQ_DEFAULT_PASS: docker
+        ports:
+          - 5672:5672
+          - 15672:15672
+    strategy:
+      matrix:
+        include: ${{ fromJSON(needs.setup-matrix.outputs.matrix-json) }}
+    steps:
+      - name: Setup
+        uses: OpenPhone/gha/.github/actions/setup@v5
+        with:
+          token: ${{ secrets.token }}
+          working-directory: ${{inputs.working-directory}}
+      # Setup of the global postgres database.
+      - name: actions/checkout@v3 @openphone/database
+        uses: actions/checkout@v3
+        if: ${{ inputs.setup_global_postgres && !inputs.skip_migration }}
+        with:
+          repository: 'OpenPhone/database'
+          ref: ${{ inputs.global_postgres_ref }}
+          path: 'database'
+          token: ${{ secrets.token }}
+
+      - name: migrate global postgres database
+        run: |
+          npm ci
+          npm run up
+        if: ${{ inputs.setup_global_postgres && !inputs.skip_migration }}
+        working-directory: 'database'
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.token }}
+          ENVIRONMENT: localdev
+          POSTGRES_HOST: ${{ inputs.postgres_host }}
+      # Run migrations against the local postgres database.
+      - name: Migrate Voice Database
+        run: ${{ inputs.migration_command }}
+        if: ${{ !inputs.setup_global_postgres && !inputs.skip_migration }}
+        env:
+          ENVIRONMENT: test
+          POSTGRES_HOST: ${{ inputs.postgres_host }}
+      # Test all the things!!!
+      - name: Run Integration Tests
+        run: ${{ inputs.test_command }} -- -- --shard=${{ matrix.number }}/${{ strategy.job-total }}
```

Comment:
Since we have a matrix action that crafts the shard array from the number of matrix input. 
Would we want to have `--shard=${{matrix.shard}}`?
where `matrix.shard` above would be `matrix.shard=[1/3, 2/3, 3/3]`